tests/NHSD.BuyingCatalogue.Identity.Api.UnitTests/Controllers/AccountControllerTests.cs

```diff
@@ -1,18 +1,198 @@
 ﻿using System;
+using System.Linq;
 using System.Threading.Tasks;
 using FluentAssertions;
 using IdentityServer4.Models;
 using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.ModelBinding;
 using Moq;
+using NHSD.BuyingCatalogue.Identity.Api.Controllers;
 using NHSD.BuyingCatalogue.Identity.Api.Services;
 using NHSD.BuyingCatalogue.Identity.Api.UnitTests.Builders;
+using NHSD.BuyingCatalogue.Identity.Api.ViewModels.Account;
 using NUnit.Framework;
+using SignInResult = NHSD.BuyingCatalogue.Identity.Api.Services.SignInResult;
 
 namespace NHSD.BuyingCatalogue.Identity.Api.UnitTests.Controllers
 {
     [TestFixture]
-	public sealed class AccountControllerTests
-	{
+    [Parallelizable(ParallelScope.All)]
+    public sealed class AccountControllerTests
+    {
+        [Test]
+        public void Error_ReturnsExpectedView()
+        {
+            using var controller = new AccountControllerBuilder().Build();
+
+            var result = controller.Error() as ViewResult;
+
+            Assert.NotNull(result);
+            result.ViewName.Should().Be("Error");
+        }
+
+        [Test]
+        public async Task Login_LoginViewModel_FailedSignIn_AddsValidationError()
+        {
+            using var controller = new AccountControllerBuilder()
+                .WithSignInResult(new SignInResult(false))
+                .Build();
+
+            await controller.Login(new LoginViewModel());
+
+            var modelState = controller.ModelState;
+
+            modelState.IsValid.Should().BeFalse();
+            modelState.Count.Should().Be(1);
+
+            (string key, ModelStateEntry entry) = modelState.First();
+            key.Should().Be(string.Empty);
+            entry.Errors.Count.Should().Be(1);
+            entry.Errors.First().ErrorMessage.Should().Be(AccountController.SignInErrorMessage);
+        }
+
+        [Test]
+        public async Task Login_LoginViewModel_FailedSignIn_ReturnsExpectedView()
+        {
+            using var controller = new AccountControllerBuilder()
+                .WithSignInResult(new SignInResult(false))
+                .Build();
+
+            var result = await controller.Login(new LoginViewModel()) as ViewResult;
+
+            Assert.NotNull(result);
```

Comment:
Any reason for the different assertion here? For consistency reasons I would try stick with the fluent assertions syntax which would be something like `result.Should().NotBeNull();`. This library tends to provide better test failure messages.

Reply:
R# currently isn't clever enough to identify the fluent version as a not null assertion. Therefore, I decided to use the `Assert.NotNull` version to avoid generating and suppressing additional R# warnings.